pkg/virt-controller/services/template.go

```diff
@@ -1385,9 +1386,29 @@ func getVirtiofsCapabilities() []k8sv1.Capability {
 	}
 }
 
+func requireDHCP(vmi *v1.VirtualMachineInstance) bool {
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if iface.Bridge != nil || iface.Masquerade != nil {
```

Comment:
There's an extra subtlety at play here; bridge bindings do **not** start the DHCP server when IPAM is enabled on the interface: https://github.com/kubevirt/kubevirt/blob/739b47753b42b30d4b9f2307d06b274579be4538/pkg/virt-launcher/virtwrap/network/podinterface.go#L454

When that happens, there's no need to explicitly request the capability.

I am not sure if it is feasible to retrieve this info at this stage in the code (when specifying the pod template) though.

Reply:
I know about this case but I don't think we can know this information in virt-controller. Please correct me if I am wrong.

Reply:
My point exactly; afaiu, we get that info by reading the IPs from the interface via netlink. Such a thing is clearly out of place in virt-controller.

From my perspective, we can assume loading the capability whenever we have a bridge interface (at least I don't know how to get that info without reading the interface via netlink). 

I think this also deserves to be detailed in the commit msg: e.g. all bridge interfaces - with or without IPAM - will explicitly have the NET_BIND_SERVICE capability.

Reply:
I updated the commit message.